cli/medperf/entities/interface.py

```diff
@@ -1,77 +1,215 @@
+        """
+        raise NotImplementedError
 
-    @abstractmethod
-    def get(cls, uid: Union[str, int]) -> "Entity":
+    @classmethod
+    def get(cls, uid: Union[str, int], local_only: bool = False) -> "Entity":
         """Gets an instance of the respective entity.
         Wether this requires only local read or remote calls depends
         on the implementation.
 
         Args:
             uid (str): Unique Identifier to retrieve the entity
+            local_only (bool): If True, the entity will be retrieved locally
 
         Returns:
             Entity: Entity Instance associated to the UID
         """
 
-    @abstractmethod
-    def todict(self) -> Dict:
-        """Dictionary representation of the entity
+        if not str(uid).isdigit() or local_only:
+            return cls.__local_get(uid)
+        return cls.__remote_get(uid)
+
+    @classmethod
+    def __remote_get(cls, uid: int) -> "Entity":
+        """Retrieves and creates an entity instance from the comms instance.
+
+        Args:
+            uid (int): server UID of the entity
 
         Returns:
-            Dict: Dictionary containing information about the entity
+            Entity: Specified Entity Instance
         """
+        logging.debug(f"Retrieving {cls.get_type()} {uid} remotely")
+        comms_func = cls.get_comms_retriever()
+        entity_dict = comms_func(uid)
+        entity = cls(**entity_dict)
+        entity.write()
+        return entity
 
-    @abstractmethod
-    def write(self) -> str:
-        """Writes the entity to the local storage
+    @classmethod
+    def __local_get(cls, uid: Union[str, int]) -> "Entity":
+        """Retrieves and creates an entity instance from the local storage.
+
+        Args:
+            uid (str|int): UID of the entity
 
         Returns:
-            str: Path to the stored entity
+            Entity: Specified Entity Instance
         """
+        logging.debug(f"Retrieving {cls.get_type()} {uid} locally")
+        entity_dict = cls.__get_local_dict(uid)
+        entity = cls(**entity_dict)
+        return entity
 
-    @abstractmethod
-    def display_dict(self) -> dict:
-        """Returns a dictionary of entity properties that can be displayed
-        to a user interface using a verbose name of the property rather than
-        the internal names
+    @classmethod
+    def __get_local_dict(cls, uid: Union[str, int]) -> dict:
+        """Retrieves a local entity information
+
+        Args:
+            uid (str): uid of the local entity
 
         Returns:
-            dict: the display dictionary
+            dict: information of the entity
         """
+        logging.info(f"Retrieving {cls.get_type()} {uid} from local storage")
+        storage_path = cls.get_storage_path()
+        metadata_filename = cls.get_metadata_filename()
+        bmk_file = os.path.join(storage_path, str(uid), metadata_filename)
```

Comment:
Just a small nitpick: I believe `bmk` means `benchmark` while here it's not necessarily a benchmark anymore:)